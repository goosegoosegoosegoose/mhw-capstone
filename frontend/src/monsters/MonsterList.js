import React, {useState, useEffect} from "react";
import MhwApi from "../api";
import MonsterCard from "./MonsterCard";
import SearchForm from "../common/SearchForm";
import { Button } from "react-bootstrap";

const MonsterList = ({add, remove}) => {
  const [monsters, setMonsters] = useState([]);

  useEffect(function fetchMonstersWhenMounted () {
    async function fetchMonsters() {
      const res = await MhwApi.request("monsters");
      setMonsters(res);
    }
    fetchMonsters();
  }, []);

  const handleClick = async (evt) => {
    const res = await MhwApi.search("monsters", evt.target.dataset.header, evt.target.dataset.handle);
    getMonsters(res);
  }

  const getMonsters = (res) => {
    setMonsters([...res])
  };

  const filters = (
    <div>
      <div>
        Size:
        <Button className="m-1" data-header="type" data-handle="small" type="button" variant="dark" size="sm" onClick={handleClick}>Small</Button>
        <Button className="m-1" data-header="type" data-handle="large" type="button" variant="dark" size="sm" onClick={handleClick}>Large</Button>
      </div>
      <div>
        Species:
        <Button className="m-1" data-header="species" data-handle="bird wyvern" type="button" variant="dark" size="sm" onClick={handleClick}>Bird Wyverns</Button>
        <Button className="m-1" data-header="species" data-handle="brute wyvern" type="button" variant="dark" size="sm" onClick={handleClick}>Brute Wyverns</Button>
        <Button className="m-1" data-header="species" data-handle="fanged wyvern" type="button" variant="dark" size="sm" onClick={handleClick}>Fanged Wyverns</Button>
        <Button className="m-1" data-header="species" data-handle="flying wyvern" type="button" variant="dark" size="sm" onClick={handleClick}>Flying Wyverns</Button>
        <Button className="m-1" data-header="species" data-handle="piscine wyvern" type="button" variant="dark" size="sm" onClick={handleClick}>Piscine Wyverns</Button>
        <Button className="m-1" data-header="species" data-handle="fanged beast" type="button" variant="dark" size="sm" onClick={handleClick}>Fanged Beast</Button>
        <Button className="m-1" data-header="species" data-handle="elder dragon" type="button" variant="dark" size="sm" onClick={handleClick}>Elder Dragons</Button>
        <Button className="m-1" data-header="species" data-handle="relict" type="button" variant="dark" size="sm" onClick={handleClick}>Relicts</Button>
        <Button className="m-1" data-header="species" data-handle="herbivore" type="button" variant="dark" size="sm" onClick={handleClick}>Herbivores</Button>
        <Button className="m-1" data-header="species" data-handle="wingdrake" type="button" variant="dark" size="sm" onClick={handleClick}>Wingdrakes</Button>
        <Button className="m-1" data-header="species" data-handle="neopteron" type="button" variant="dark" size="sm" onClick={handleClick}>Neopterons</Button>
        <Button className="m-1" data-header="species" data-handle="fish" type="button" variant="dark" size="sm" onClick={handleClick}>Fish</Button>
      </div>
    </div>
  )

  if (monsters.length === 0) {
    return (
      <div className="container text-center">
        <h1>Monsters</h1>
        <SearchForm type="monsters" header="name" get={getMonsters} />
        {filters}
        <p>Sorry, no results were found! Please re-submit to refresh.</p>
      </div>
    )
  }

  return (
    <div className="container text-center">
      <h1>Monsters</h1>
      <SearchForm type="monsters" header="name" get={getMonsters} />
      {filters}
      <div className="row justify-content-center">
        {monsters.map(m => 
          <MonsterCard
            key={m.id}
            id={m.id}
            name={m.name}
            icon={m.icon}
            add={add}
            remove={remove}
          />
        )}
      </div>
    </div>
  )
};


export default MonsterList;