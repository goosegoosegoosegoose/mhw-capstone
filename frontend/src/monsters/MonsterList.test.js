import { render } from '@testing-library/react';
import MonsterList from "./MonsterList";

test('smoke test', () => {
  render(      
    <MonsterList />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<MonsterList />);
  expect(asFragment()).toMatchSnapshot();
});