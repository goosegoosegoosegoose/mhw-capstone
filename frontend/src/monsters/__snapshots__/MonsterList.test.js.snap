// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="container text-center"
  >
    <h1>
      Monsters
    </h1>
    <form>
      <div
        class="my-3 row"
      >
        <div
          class="input-group col-2"
        >
          <input
            class="form-control"
            placeholder="Enter search"
            value=""
          />
          <button
            class="mx-1 btn btn-dark btn-sm"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
    <div>
      <div>
        Size:
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="small"
          data-header="type"
          type="button"
        >
          Small
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="large"
          data-header="type"
          type="button"
        >
          Large
        </button>
      </div>
      <div>
        Species:
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="bird wyvern"
          data-header="species"
          type="button"
        >
          Bird Wyverns
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="brute wyvern"
          data-header="species"
          type="button"
        >
          Brute Wyverns
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="fanged wyvern"
          data-header="species"
          type="button"
        >
          Fanged Wyverns
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="flying wyvern"
          data-header="species"
          type="button"
        >
          Flying Wyverns
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="piscine wyvern"
          data-header="species"
          type="button"
        >
          Piscine Wyverns
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="fanged beast"
          data-header="species"
          type="button"
        >
          Fanged Beast
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="elder dragon"
          data-header="species"
          type="button"
        >
          Elder Dragons
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="relict"
          data-header="species"
          type="button"
        >
          Relicts
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="herbivore"
          data-header="species"
          type="button"
        >
          Herbivores
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="wingdrake"
          data-header="species"
          type="button"
        >
          Wingdrakes
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="neopteron"
          data-header="species"
          type="button"
        >
          Neopterons
        </button>
        <button
          class="m-1 btn btn-dark btn-sm"
          data-handle="fish"
          data-header="species"
          type="button"
        >
          Fish
        </button>
      </div>
    </div>
    <p>
      Sorry, no results were found! Please re-submit to refresh.
    </p>
  </div>
</DocumentFragment>
`;
